File: packages/flyto-indexer/tests/fixtures/vuln/taint_regex.js
// Single-line source → sink fixtures to exercise the JS regex fallback
// in analyzer/taint_rules.py (JS_TAINT_PATTERNS).

// SSRF
app.get('/p', async (req, res) => { const r = await axios.get(req.query.u); res.send(r.data) })

// Open redirect
app.get('/g', (req, res) => { res.redirect(req.query.next) })

// NoSQL injection
app.post('/s', async (req, res) => { const u = await Model.find(req.body); res.json(u) })

// CRLF / header injection
app.post('/h', (req, res) => { res.setHeader('X-User', req.body.name) })

// ReDoS — constructing RegExp from user input
app.get('/m', (req, res) => { const re = new RegExp(req.query.pat) })

// Prototype pollution
app.post('/merge', (req, res) => { Object.assign({}, req.body) })

File: packages/flyto-indexer/tests/fixtures/vuln/taint_regex.go
// Single-line source → sink fixtures exercising GO_TAINT_PATTERNS.
package main

import (
	"encoding/xml"
	"net/http"
	"regexp"
	"github.com/gin-gonic/gin"
)

// SSRF via Gin source
func ssrf(c *gin.Context) { http.Get(c.Query("u")) }

// Open redirect
func openRed(c *gin.Context, w http.ResponseWriter, r *http.Request) {
	http.Redirect(w, r, c.Query("next"), http.StatusFound)
}

// CRLF — Gin source into header
func crlf(c *gin.Context, w http.ResponseWriter) { w.Header().Set("X-User", c.Query("name")) }

// ReDoS — regex compile with user input
func redos(c *gin.Context) { regexp.Compile(c.Query("pat")) }

// SQLi — Gin source into db.Query
func sqli(c *gin.Context, db *DB) { db.Query("SELECT * FROM t WHERE id=" + c.Query("id")) }

// XXE — user body into xml.NewDecoder
func xxe(r *http.Request) { xml.NewDecoder(r.Body).Decode(nil) }
